src/main/java/org/spongepowered/api/schematic/Schematic.java

```diff
@@ -0,0 +1,210 @@
+
+    /**
+     * Gets the {@link BlockState} at the relative location, or {@link Optional#empty()} if none.
+     *
+     * @param extent The extent
+     * @param location The location (relative to the origin of the schematic) of the BlockSnapshot desired
+     * @return The BlockState found
+     */
+    Optional<BlockState> getBlockAt(Extent extent, Vector3i location);
+
+    /**
+     * Gets all {@link BlockState}s in the Schematic.
+     *
+     * @return All BlockStates
+     */
+    BlockState[] getBlocks();
+
+    /**
+     * Gets all {@link SchematicEntity} in the Schematic.
+     *
+     * @return All entities
+     */
+    SchematicEntity[] getEntities();
+
+    /**
+     * Gets all TileEntities in the Schematic.
+     *
+     * @return All tile entities
+     */
+    SchematicTileEntity[] getTileEntities();
+
+
+    interface Builder extends DataBuilder<Schematic> {
+
+
+        /**
+         * Adds the given {@link DataContainer} to the schematic's metadata.
+         *
+         * @param metadata The metadata to add
+         * @return This schematic builder
+         */
+        Builder metadata(DataContainer metadata);
+
+        /**
+         * Sets the minimum point of the schematic.
+         *
+         * @param point The point to set
+         * @return This schematic builder
+         */
+        Builder minimumPoint(Vector3i point);
+
+        /**
+         * Sets the maximum point of the schematic.
+         *
+         * @param point The point to set
+         * @return This schematic builder
+         */
+        Builder maximumPoint(Vector3i point);
+
+        /**
+         * Sets the origin of the schematic.
+         *
+         * <p>Correlates to the `OffsetX`, `OffsetY`, and `OffsetZ` fields of the Schematic Specification (v1).</p>
+         *
+         * @param origin The origin point to set
+         * @return This schematic builder
+         */
+        Builder origin(Vector3i origin);
+
+        /**
+         * Adds a {@link BlockState} to the palette. Duplicate BlockStates will be ignored.
+         *
+         * @param blockState
+         * @return This schematic builder
+         */
+        Builder addToPalette(BlockState blockState);
+
+        /**
+         * Should offer automated conversions between versions.
+         *
+         * @param holder
+         * @return This schematic builder
+         */
+        Builder from(Schematic holder);
+
+        /**
+         * Builds the schematic object
+         *
+         * @return The built Schematic object
+         * @throws InvalidDataException if any required fields are not completed
+         */
+        Optional<Schematic> build() throws InvalidDataException;
+
+        /**
+         * Builds the schematic object using locations from the given world
+         *
+         * @param world
+         * @return
+         * @throws InvalidDataException
+         */
+        Optional<Schematic> build(World world) throws InvalidDataException;
```

Comment:
the build method of builders should never return Optional so kashike is right. This will be fixed elsewhere in the API.